### PR TITLE
Remove the unused files and restructure for ease-of-use

### PR DESCRIPTION
**Pull request by kmova**
This PR contains a list of changes that help in cleaning up the openebs/openebs/k8s folder and changes to the CAS templates to make it backward compatible with 0.6. The changes include:
(a) Move the unused/developement folder - crd-samples and openebs-monitoring. They are moved to elves and community repo respectively. 
(b) Remove the openebs-kubernetes-dashboard.yaml
(c) Remove the storagepool and storagepoolclaims files and merge their content into config.yaml 
(d) Update the jiva cas templates to match the deployments generated in 0.6

### CODE REVIEW

File: k8s/openebs-config.yaml [hunk 1]
```diff
@@ -22,3 +22,43 @@ metadata:
   type: hostdir
 spec:
   path: "/var/openebs"
+---
```
Comment (AmitKumarDas) [inline k8s/openebs-config.yaml]: StoragePool is duplicated. Is it ok?

File: k8s/README.md [hunk 2]
```diff
@@ -1,35 +1,31 @@
 # Using OpenEBS with K8s
 
-This folder contains the software components (plugins, scripts, drivers etc.,), usage instructions and application examples of using OpenEBS with K8s. 
+This folder contains the artifacts (YAMLS, plugins, scripts, drivers etc.,), usage instructions and application examples of using OpenEBS with K8s. The artifiacts in this repository contain unreleased changes. If you are looking at deploying from a stable release, please follow the instructions at [Quick Start Guide](https://docs.openebs.io/docs/next/quickstartguide.html)
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/README.md]: I reworded the paragraph. do let me know if the new edits help. I agree it is important to convey that this is only for developer DIY stuff.

File: k8s/README.md [hunk 3]
```diff
@@ -1,35 +1,31 @@
...
 ## Usage
 
-### Enable OpenEBS on K8s
+### Installing OpenEBS on K8s
 ```
 kubectl apply -f openebs-operator.yaml
 ```
 
-### Customize OpenEBS
+There is work under way to install/load the cas templates by maya-apiserver on startup. Till then, need to manually load the cas templates.
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/README.md]: Good point. Made some changes and also renamed the file.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 4]
```diff
@@ -1161,28 +1161,18 @@ metadata:
   name: jiva-volume-create-default-0.7.0
 spec:
   defaultConfig:
-  - name: VolumeMonitor
-    enabled: "true"
   - name: ControllerImage
     value: openebs/jiva:0.6.0
   - name: ReplicaImage
     value: openebs/jiva:0.6.0
+  - name: ExporterImage
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: cool. changed it.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 5]
```diff
@@ -1472,11 +1492,13 @@ data:
     metadata:
       labels:
         openebs/controller-service: jiva-controller-service
-        openebs.io/controller-service: jiva-controller-svc
-        openebs.io/storage-engine-type: jiva
         openebs/volume-provisioner: jiva
         vsm: {{ .Volume.owner }}
-        openebs.io/pv: {{ .Volume.owner }}
+        pvc: {{ .Volume.pvc }}
+        openebs.io/storage-engine-type: jiva
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: sticking with storage-engine-type for now. @avishnu @AmitKumarDas Will change it a separate PR if new convention is decided.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 6]
```diff
@@ -1472,11 +1492,13 @@ data:
...
+        openebs.io/controller-service: jiva-controller-svc
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: Same as above. leaving it as `controller-service` for now. But change it to `storage-target-service`.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 7]
```diff
@@ -1656,6 +1687,9 @@ data:
             - launch
             image: {{ .Config.ControllerImage.value }}
             name: {{ .Volume.owner }}-ctrl-con
+            env:
+            - name: "REPLICATION_FACTOR"
+              value: {{ .Config.ReplicaCount.value }}
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: No. Replication factor is in fact the replica count.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 8]
```diff
@@ -1665,8 +1699,8 @@ data:
           - args:
             - -c=http://127.0.0.1:9501
             command:
-            - maya-volume-exporter
-            image: {{ $monitorVal }}
+            - maya-exporter
+            image: {{ .Config.ExporterImage.value }}
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: changed it.

File: k8s/openebs-cas-templates-pre-alpha.yaml [hunk 9]
```diff
@@ -1697,42 +1739,44 @@ data:
...
         vsm: {{ .Volume.owner }}
-        openebs.io/pv: {{ .Volume.owner }}
+        pvc: {{ .Volume.pvc }}
+        openebs.io/storage-engine-type: jiva
+        openebs.io/replica: jiva-replica
+        openebs.io/persistentvolume: {{ .Volume.owner }}
+        openebs.io/persistentvolumeclaim: {{ .Volume.pvc }}
+      annotations:
         openebs.io/capacity: {{ .Volume.capacity }}
+        openebs.io/storage-pool: {{ .Config.StoragePool.value }}
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/openebs-cas-templates-pre-alpha.yaml]: going with `storage-pool` as discussed to stick with names split by - .

File: k8s/openebs-config.yaml (hunk 1 above)
Comment (kmova) [reply k8s/openebs-config.yaml]: yes.

File: k8s/openebs-pre-release-features.yaml [hunk 10]
```diff
@@ -1161,28 +1161,18 @@ metadata:
   name: jiva-volume-create-default-0.7.0
 spec:
   defaultConfig:
-  - name: VolumeMonitor
-    enabled: "true"
   - name: ControllerImage
     value: openebs/jiva:0.6.0
   - name: ReplicaImage
     value: openebs/jiva:0.6.0
+  - name: VolumeMonitorImage:ci
```
Comment (AmitKumarDas) [inline k8s/openebs-pre-release-features.yaml]: Invalid name

Comment (kmova) [reply k8s/openebs-pre-release-features.yaml]: thanks for catching this @AmitKumarDas. Fixed it.

**Pull request merged**